src/lua/tags.c

```diff
@@ -148,7 +149,30 @@ static int tag_delete(lua_State *L)
 {
   dt_lua_tag_t tagid;
   luaA_to(L, dt_lua_tag_t, &tagid, -1);
-  dt_tag_remove(tagid, true);
+  
+  GList *tagged_images = NULL;
+  sqlite3_stmt *stmt;
+  DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), "SELECT imgid FROM main.tagged_images WHERE tagid=?1",
+                              -1, &stmt, NULL);
+  DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, tagid);
+  while(sqlite3_step(stmt) == SQLITE_ROW)
+  {
+    tagged_images = g_list_append(tagged_images, GINT_TO_POINTER(sqlite3_column_int(stmt, 0)));
+  }
+  sqlite3_finalize(stmt);
+
+  dt_tag_remove(tagid, TRUE);
+
+  GList *list_iter;
+  if((list_iter = g_list_first(tagged_images)) != NULL)
+  {
+    do
+    {
+      dt_image_synch_xmp(GPOINTER_TO_INT(list_iter->data));
+    } while((list_iter = g_list_next(list_iter)) != NULL);
+  }
+  g_list_free(g_list_first(tagged_images));
+
```

Comment:
The code is from src/libs/tagging.c, delete_button_clicked. dt_tag_remove only removes the tag from the database. The XMPs are not updated. The code collects all images with the tag before the tag is removed and syncs the XMPs after the tag is removed.